app/src/main/java/in/testpress/testpress/ui/MainActivity.java

```diff
@@ -186,21 +196,83 @@ public boolean onLongClick(View view) {
         });
     }
 
+    private void initializeNavigationHeaderView() {
+        View headerView = navigationView.getHeaderView(0);
+        navigationHeader = headerView.findViewById(R.id.navigation_header_container);
+        username = headerView.findViewById(R.id.tv_profile);
+        profilePicture = headerView.findViewById(R.id.img_profile);
+    }
+
+    private void getProfileDetailsFromViewModel() {
+        if (isUserAuthenticated) {
+            profileDetailsViewModel.get(serviceProvider, this).observe(this, new Observer<ProfileDetails>() {
```

Comment:
Don't need to pass serviceProvider. Create repository and inject it in constructor